### DISCUSSION TITLE
[FEATURE] New MinIO-Zig Client Library

### DISCUSSION THREAD

**Discussion by shanduur** [minio/minio#20519] [Q&A]
**Is your feature request related to a problem? Please describe.**
Currently, MinIO client libraries exist for multiple programming languages, but there is no official or community-supported client library for the Zig programming language. Developers who prefer Zig have to manually implement MinIO APIs, which increases development time and introduces potential inconsistencies or bugs. This is particularly frustrating for projects using Zig, which would benefit from an optimized and native client library for MinIO.

**Describe the solution you'd like**
A native MinIO client library written in Zig, allowing seamless interaction with MinIO object storage from Zig applications. This library should support basic MinIO operations like bucket management, object CRUD (Create, Read, Update, Delete), pre-signed URLs, and proper handling of streaming for large files. It should leverage Zig’s low-level control and performance benefits, ensuring efficient memory usage and error handling, while also being idiomatic to the language.

**Describe alternatives you've considered**
- Using the MinIO client libraries written in other languages (Go, Python, etc.) through FFI (Foreign Function Interface). However, this introduces complexity and may not yield optimal performance.
- Manually implementing the required functionality in Zig for specific projects. This, however, duplicates effort and risks inconsistencies across different Zig-based projects.
- Relying on AWS S3-compatible libraries for Zig, e.g. [aws-sdk-for-zig](https://git.lerch.org/lobo/aws-sdk-for-zig), though they are often fat compared to MinIO clients.

**Additional context**
The Zig programming language is gaining popularity due to its performance, simplicity, and control over system resources, making it a great fit for applications interacting with object storage like MinIO. A dedicated MinIO client library for Zig would help expand MinIO’s reach into more systems and environments.

**Comment by harshavardhana** [comment]
How is it better than minio-cpp or minio-rs libraries?
